doc/chapters/api/index.rst

```diff
@@ -1,111 +1,163 @@
-
 .. _api:
 
 API
 ***
 
-This chapter documents the DBus and C API for interacting with the SoftwareContainerAgent. This is
+This chapter documents the D-Bus API for interacting with the SoftwareContainerAgent. This is
 mostly useful for integrators.
 
-.. TODO::
-    This chapter is still to be written properly. Some bits and pieces have
-    been but they are under development.
-
 .. _dbus-api:
 
-DBus API
+D-Bus API
 ========
 
-:Path: /com/pelagicore/SoftwareContainer
+D-Bus API is an IPC interface to call SoftwareContainer agent methods. The API provides following object path and interface.
+
+:Object Path: /com/pelagicore/SoftwareContainer
 :Interface: com.pelagicore.SoftwareContainerAgent
 
 Methods
 -------
 
 Ping
 ----
-Ping()
+Controls availability of com.pelagicore.SoftwareContainerAgent interface
+
+:Parameters:
+        *None*
+|
+:Return Value:
+        *None*
 
 CreateContainer
 ---------------
-:containerIDPrefix: string
-:config: string
-
-uint32 containerID CreateContainer(string containerIDPrefix, string config)
+Creates a new container and returns created container id.
+
+:Parameters:
+        :IDPrefix: ``string`` prefix for name of back-end container.
+        :config: ``string`` config file in json format.
+        
+                Example config JSON::
+        
+                [{"enableWriteBuffer": true}]
+|
+:Return Value:
+        :containerID: ``uint32`` ID of created SoftwareContainer.
 
 SetContainerName
 ----------------
-:containerID: uint32
-:containerName: string
+Sets the name of container with unique containerID.
 
-void SetContainerName(uint32 containerID, string containerName)
+:Parameters:
+        :containerID: ``uint32`` The ID obtained by CreateContainer method.
+        :containerName: ``string`` name.
+|
+:Return Value:
+        *None*
 
 LaunchCommand
 -------------
-:containerID: uint32
-:userID: uint32
-:commandLine: string
-:workingDirectory: string
-:outputFile: string
-:env: map<string, string>
-
-uint32 pid LaunchCommand(uint32 containerID, uint32 userID, string commandLine, string workingDirectory, string outputFile, map<string, string> env)
+Launches specific application/code in the container.
```

Comment:
I think "specific" should be changed, perhaps write "Launches the specified ..."